WordPress/gutenberg
# BorderBoxControl: Refactor stories to TypeScript and Controls

Part of: https://github.com/WordPress/gutenberg/issues/35744

## What?
Finishes off conversion of BorderBoxControl to TypeScript, including updating stories to use controls instead of knobs.

## Why?
- To autogenerate docs in Storybook.

## How?
- Update export of `BorderBoxControl` named component so the type data can be extracted.
- Marks `BorderBoxControl` as non-polymorphic.
- Adds usage info and example from README as main JSDoc for the component.
- Converts stories to TypeScript.
- Removed the "default border" value as it added unwarranted complexity with the current approach to these stories.

## Testing Instructions

1. `npm run storybook:dev`
2. Check the [`BorderBoxControl` story](http://localhost:50240/?path=/story/components-experimental-borderboxcontrol--default) still function as expected
3. Check the generated docs.
